packages/common/http/src/params.ts

```diff
@@ -73,7 +73,7 @@ export class HttpUrlEncodingCodec implements HttpParameterCodec {
 function paramParser(rawParams: string, codec: HttpParameterCodec): Map<string, string[]> {
   const map = new Map<string, string[]>();
   if (rawParams.length > 0) {
-    const params: string[] = rawParams.split('&');
+    const params: string[] = rawParams.replace(/^\?/, '').split('&');
```

Comment:
I'd propose adding a quick comment here explaining why we drop the leading `?` char:
```suggestion
    // The `window.location.search` can be used while creating an instance of the `HttpParams` class (e.g.
    // `new HttpParams({ fromString: window.location.search })`). The `window.location.search` may start with
    // the `?` char, so we strip it if it's present.
    const params: string[] = rawParams.replace(/^\?/, '').split('&');
```